src/robot.js

```diff
@@ -21,11 +22,31 @@ class Robot {
   // Robots receive messages from a chat source (Campfire, irc, etc), and
   // dispatch them to matching listeners.
   //
-  // adapterPath -  A String of the path to built-in adapters (defaults to src/adapters)
-  // adapter     - A String of the adapter name.
-  // httpd       - A Boolean whether to enable the HTTP daemon.
-  // name        - A String of the robot name, defaults to Hubot.
-  constructor (adapterPath, adapter, httpd, name, alias) {
+  // @param {Object} params - parameters for building Robot
+  // @param {String} params.adapterPath -  A String of the path to built-in adapters (defaults to src/adapters)
+  // @param {String} params.adapter     - A String of the adapter name.
+  // @param {String} params.httpd       - A Boolean whether to enable the HTTP daemon.
+  // @param {String} params.name        - A String of the robot name, defaults to Hubot.
+  constructor (params) {
+    var adapterPath, adapter, httpd, name, alias
+    if (arguments.length > 1) {
+      adapterPath = arguments[0]
+      adapter = arguments[1]
+      httpd = arguments[2]
+      name = arguments[3]
+      alias = arguments[4]
+
+      deprecate(`Robot constructed with multiple arguments, construct with one object argument instead`)
```

Comment:
just a minor style comment, I would use single quotes for strings unless you use place holders in it.